server/services/calculateReleaseDatesService.ts

```diff
@@ -59,23 +64,127 @@ export default class CalculateReleaseDatesService {
   ): Promise<{
     calculationBreakdown?: CalculationBreakdown
     effectiveDates?: { [key: string]: DateBreakdown }
+    releaseDatesWithAdjustments: ReleaseDateWithAdjustments[]
   }> {
     try {
       const breakdown = await this.getCalculationBreakdown(calculationRequestId, token)
       return {
         calculationBreakdown: breakdown,
         effectiveDates: this.getEffectiveDates(releaseDates, breakdown),
+        releaseDatesWithAdjustments: this.extractReleaseDatesWithAdjustments(breakdown),
       }
     } catch (error) {
       // If an error happens in this breakdown, still display the release dates.
       logger.error(error)
       return {
         calculationBreakdown: null,
         effectiveDates: null,
+        releaseDatesWithAdjustments: null,
       }
     }
   }
 
+  // TODO expand this functionality to retrieve the 'other key dates with adjustments' based on the rules from the
```

Comment:
Will follow up with a second ticket to fetch the other release date types from the backend in the same manner.. at the moment they are derived on the UI side using the sentences